glossary.rst

```diff
@@ -0,0 +1,151 @@
+
+.. _glossary:
+
+Glossary
+========
+
+.. glossary::
+   :sorted:
+
+   conanfile
+      Can refer to either `conanfile.txt` or `conanfile.py` depending on what's the context it is
+      used in.
+
+   conanfile.py
+      The file that defines a Conan recipe that can be used to create or consume packages. Inside of
+      this recipe, it is defined (among other things) how to download the package's source code, how to
+      build the binaries from those sources, how to package the binaries and information for futures
+      consumers on how to consume the package.
+
+   conanfile.txt
+      The file used to define a package or list of packages to be consumed by a project. It can define
+      some more auxiliary files to build or run the project such as generators for different build
+      systems or files to be imported from the package to the local folder.
+
+   recipe
+      Python script defined in a `conanfile.py` that specifies how the package is built from sources,
+      what the final binary artifacts are, the package dependencies, etc.
+
+   reference
+      A package reference is the combination of the package name, version, and two optional fields
+      named user and channel that could be useful to identify a forked recipe from the community with
+      changes specific to your company. It is show in the the form of: `<name>/<version>@<user>/<channel>`.
+
+   package
+      A Conan package is a collection of files meant to be consumed for a certain configuration and
+      settings. It can contain binary files such as libraries, headers or tools to be reused by the
+      consumer of the package.
+
+   lockfile
+      Files that store the information with the exact versions, revisions, options, and configuration
+      of a dependency graph. They are intended to make the building process reproducible even if the
+      dependency definitions in conanfile recipes are not fully deterministic.
+
+   revision
+      Is the way to implicitly version the changes done in a recipe or package without bumping the
+      actual reference or package version.
+
+   recipe revision
+      A unique ID using the latest VCS hash or a checksum of the recipe manifest (`conanfile.py` with
+      files exported if any).
+
+   package revision
+      A unique ID using the checksum of the package manifest (all files stored in a binary package).
+
+   package ID
+      The package id is a hash of the settings options and requirements used to identify the binary
+      packages
```

Comment:
```suggestion
      packages. Applying different profiles to the `conan create` command, it will generate different package IDs. e.g: Windows, x86, shared...
```